### DISCUSSION TITLE
How can I submit a transaction to all endorsing peers of an org?

### DISCUSSION THREAD

**Discussion by bh4rtp** [hyperledger/fabric-gateway#608] [Q&A] [answered]
In the `fabric-node-sdk`, I can do it by `endorsers = channel.getEndorsers('Org1')` and `transaction.setEndorsingPeers(endorsers)` .
How can I do it using `fabric-gateway node`?

**Comment by bestbeforetoday** [comment, marked as answer]
You don't need to. The Gateway service running in the peer selects the most up-to-date set endorsing peers from the organisations required to meet the effective endorsement policy for a specific transaction invocation. Using additional (potentially less up-to-date) peers increases the overhead of the transaction invocation and increases the chances of transaction failure due to inconsistent endorsements. See the [Fabric documentation](https://hyperledger-fabric.readthedocs.io/en/release-2.5/gateway.html#how-the-gateway-endorses-your-transaction-proposal) for details.

**Reply by bh4rtp** [reply]
I give the endorsement policy as `OR(Org1, Org2)` to deploy the chaincode. When using `fabric-gateway`, every time the transaction is submitted to only peer of an organization. This sounds not be majority endorsement by that organization. How can I make `fabric-gateway` endorse like `OutOf(2, peer0.org1, peer1.org1, peer2.org1)`  if only `Org1` is choosed as endorsement organization? The client and chaincode know which function should be endorsed by `[Org1, Org2]`, `[Org1] or [ Org2]`. I want the majority of peers should act as endorser if `[Org1] or [Org2]` is enabled.

**Reply by bh4rtp** [reply]
And how can `fabric-gateway` set interest on collections like legacy node sdk?

**Reply by andrew-coleman** [reply]
As Mark said, you don't need to.  With the new gateway service (via the client APIs), you request that a transaction is submitted to the ledger and it does that for you, including all the low-level implementation specific concerns such as peer selection, private collection policies, SBE policies, etc.  
The old SDKs exposed all of those concerns to the client application which made them harder for application developers to use and harder for the maintainers to support.  So we sunk all of that logic into the peer.  The old SDKs are no longer supported.

**Reply by bh4rtp** [reply]
I just read the documentation. `Fabric-gateway` selects the peer from an organization with the highest ledger block height. Does this imply the gateway considers this peer is the preferably trusted peer of the organization?

**Reply by andrew-coleman** [reply]
It depends what you mean by 'trust'.  The gateway trusts all of the peers in the organisation from the point of view of being non-malicious.  The reason it chooses the one with the highest block height is because that peer will have the most up-to-date copy of the ledger and hence will be least likely to cause an MVCC error.

**Reply by bestbeforetoday** [reply]
To add to that, if your concern is ensuring that a single malicious peer or organisation cannot endorse a bad transaction, you might want to define an endorsement policy that requires endorsement from more than one organisation.

**Reply by bh4rtp** [reply]
@bestbeforetoday Yes, exactly I concern about a single malicious peer and will change the endorsement policy by multiple organization.
